frmts/jpegxl/jpegxl.cpp

```diff
@@ -1798,16 +1798,35 @@ GDALDataset *JPEGXLDataset::CreateCopy(const char *pszFilename,
         }
     }
 
+    constexpr int DEFAULT_EFFORT = 5;
+    const int nEffort = atoi(CSLFetchNameValueDef(
+        papszOptions, "EFFORT", CPLSPrintf("%d", DEFAULT_EFFORT)));
+    const char *pszDistance = CSLFetchNameValue(papszOptions, "DISTANCE");
+    const char *pszQuality = CSLFetchNameValue(papszOptions, "QUALITY");
+    const char *pszAlphaDistance =
+        CSLFetchNameValue(papszOptions, "ALPHA_DISTANCE");
     const char *pszLossLessCopy =
         CSLFetchNameValueDef(papszOptions, "LOSSLESS_COPY", "AUTO");
-    if (EQUAL(pszLossLessCopy, "AUTO") || CPLTestBool(pszLossLessCopy))
+    if ((EQUAL(pszLossLessCopy, "AUTO") && !pszDistance && !pszQuality &&
```

Comment:
Not sure if I read this correctly but it seems to me it can be reduced to 
if (EQUAL(pszLossLessCopy, "AUTO") && ((first group of checks) || (second group of checks))

Reply:
actually not because the second group test for *not* equal to AUTO